packages/wallet-sdk/src/connection/WalletLinkConnection.ts

```diff
@@ -416,57 +401,60 @@ export class WalletLinkConnection {
 
   private requestResolutions = new Map<IntNumber, (_: ServerMessage) => void>();
 
-  private async makeRequest<T extends ServerMessage>(
+  private async makeRequest<T extends ServerMessageType, M = ServerMessage<T>>(
```

Comment:
for example
![image](https://github.com/coinbase/coinbase-wallet-sdk/assets/966662/a11af7e6-31ef-4175-aeb3-73c1430528a1)